adafruit_ads1x15/analog_in.py

```diff
@@ -55,12 +55,49 @@ def value(self) -> int:
         Even if the underlying analog to digital converter (ADC) is
         lower resolution, the value is 16-bit.
         """
-        return self._ads.read(
-            self._pin_setting, is_differential=self.is_differential
-        ) << (16 - self._ads.bits)
+        return self._ads.read(self._pin_setting, is_differential=self.is_differential)
 
     @property
     def voltage(self) -> float:
         """Returns the voltage from the ADC pin as a floating point value."""
-        volts = self.value * _ADS1X15_PGA_RANGE[self._ads.gain] / 32767
+        volts = self.convert_to_voltage(self.value)
+        return volts
+
+    def convert_to_value(self, volts: float) -> int:
+        """Calculates 12-bit integer for threshold registers from voltage level input"""
+
+        # Convert 2's complement of signed int if number is negative
```

Comment:
Why do you need to worry about two's complement? The math should do that for you.

You can use `//` instead of `/` to do integer division and not need rounding.

I don't think you need to worry about the number of bits here either. Just do voltage to 16bits. However, maybe you want to convert to 17bit number so that you have a sign bit. (`.value` is 16-bit unsigned) Then the threshold setter can ignore the lowest bit.

Reply:
Maybe I'm wrong, but I don't think python has any way of knowing that 0xFFFF is supposed to mean -1 instead of 65535.   If you multiply those numbers by the LSB to convert to volts, you get vastly different numbers. That's why I do the two's complement.

I'll look into using //, but I'm currently using it to divide 2 floats, so I need to make sure that works right.

I still don't understand the 17-bit thing. If anything, you could just use the bit 16 as a 'sign' bit and not do two's complement. I don't get adding one bit just to ignore another.